### ISSUE TITLE
"message.createButtonCollector is not a function" when getting message via "msg.channel.fetch"

### ISSUE THREAD

**Issue by Blade67** [AngeloCore/discord-buttons#94]
Example:
```js
let message = await msg.channel.fetch(message);

const collector = message.createButtonCollector((button) => button.clicker.user.id === msg.author.id, {
    time: 5000,
});
```

Error:
```js
TypeError: message.createButtonCollector is not a function
```

I assume messages don't get casted to the patched ExtendedMessage class.

**Comment by Nagol12344** [comment]
Its telling you its not a function, its saying you can't have the () after it.

**Comment by Blade67** [comment]
Thanks Sherlock, I didn't notice it said **`is not a function`** because that totally would make sense **if it wasn't declared as function in the library**.

This is not a basic JavaScript issue, it's a library issue.

**Comment by zMelihStrqfe** [comment]
Use this:
```js
let message = await msg.channel.fetch(message).then(async (msg) => {

   const collector = msg.createButtonCollector((button) => button.clicker.user.id === msg.author.id, {
       time: 5000,
   });
});```

**Comment by Blade67** [comment]
I don't see how that would fix my issue. I'm using `async/await` instead of Promises.
I tried it to be safe (in case it was some weird JavaScript behavior) but same result.

**Comment by AngeloCore** [comment]
@Blade67 do you have this problem with other collector? Like `ReactionCollector`

**Comment by BliiTzZ** [comment]
@Blade67 You failed to fetch your message because your code is wrong.

message.channel.fetch() fetches a channel based on an id, not a message.

If you want to fetch a message in the channel you have to do message.channel.messages.fetch("message id");

for example:

`js
let message = await msg.channel.messages.fetch("message id");

const collector = message.createButtonCollector((button) => button.clicker.user.id === msg.author.id, {
    time: 5000,
});
`

**Comment by Blade67** [comment]
@BliiTzZ That's a typo on my part. I did that, but after editing a message, the ID seems to change or the message changes to the legacy type.
To try this, try editing a message in a function by passing message (or message.id and re-fetching), you'll notice some unexpected behavior.

**Comment by AlanORegan** [comment]
@Blade67 I got a similar issue whenforgetting to bind discord-buttons to the client, i.e.: the last line of:
`const discord = require('discord.js'); // Define / Require the discord.js module.
const client = new discord.Client(); // Creating a discord.js client instance (constructor).
const disbut = require('discord-buttons')(client); // Requiring discord-buttons and binding it to the initialised client.`

**Comment by AlanORegan** [comment]
@Blade67 I got a very similar error when forgetting to bind discord-buttons to the client, like the last line of this:
```javascript
const discord = require('discord.js');
const client = new discord.Client();
const disbut = require('discord-buttons');
disbut(client);
```
. . . and that worked until I tried porting the bot to Typescript, and then the error at the top of this post was back.  The version of discord.js here is v12.5.3.  Fully expect someone will have a better solution than the (**temporary**)  one below (hope they do!), but this got discord-buttons working in Typescript. 

Step 1: Add this line to node-modules/discord-buttons/src/index.js
```javascript
module.exports.ExtendedMessage = require(`./v${version}/Classes/Message`);
```
It exports the extended Message class that discord-buttons has provided.

Step 2: Change how you create the collector: 
```javascript
const msg = await message.author
    .send(`Activity: ${requestedActivity} is Ready`, {
        component: buttonRow,
    } as any) as ExtendedMessage;

const filter = (button) => button.clicker.user.id === message.author.id;
const collector = msg.createButtonCollector(
    filter,
    {
        time: 43200000, // 12 hours
    } as any,
) as any;
```
When sending the message, the "as any) as ExtendedMessage;" casts the message object returned to ExtendedMessage - so then TypeScript can see that the createButtonCollector method is available.  You will need the other two 'as any' casts on the collector creation for Typescript not to complain.

The collector.on statement is unchanged an works unchanged from raw javascript.

Hope this helps!  discord-buttons is awesome!
